datasets/ee/riigikogu/crawler.py

```diff
@@ -0,0 +1,102 @@
+from zavod import Context, helpers as h
+from zavod.logic.pep import categorise
+
+def get_members_url(context: Context) -> list:
+    """
+    Fetches the URLs for the pages of each member from the main page.
+
+    :param context: The context object.
+
+    :return: The URLs for the pages.
+    """
+
+    try:
+        main_website = context.fetch_html(context.data_url)
+
+    except:
+        context.log.error("Couldn't fetch main website")
+        return None
+
+    # this XPath corresponds to the a tags wit the links for a member page
+    xpath_to_a_tags = "//*[@id='main']/section/div[2]/div/ul/li/div/div/ul/li[1]/h3/a"
+
+    return [a_tag.get("href") for a_tag in main_website.xpath(xpath_to_a_tags)]
+
+def crawl_item(member_url: dict, context: Context):
+    """
+    Creates an entity, a position and a occupancy for each member,
+    we fetch the page of that member, extract the
+    relevant information and finally create
+    a Person, Position and Occupation.
+
+    :param member_url: The url for that member webpage
+    :param context: The context object.
+    """
+
+    HEADERS = {
+        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
+    }
+
+    member_page_html = context.fetch_html(member_url, headers=HEADERS)
+
+    try:
+        name = member_page_html.xpath('//*[@id="main"]/header/div/h1/text()')[0]
+    except:
+        context.log.error("Couldn't find name")
+        name = ""
+
+
+    entity = context.make("Person")
+    entity.id = context.make_id(name)
+
+    try:
+        # the phone number apparently is the first contact in the list
+        # but to make sure we correctly get it, we based our xpath on the icon
+        phone_number = member_page_html.xpath('//*[@class="icon icon-tel"]/../text()')[0].replace(" ", "")
+        entity.add("phone", phone_number)
+    except:
+        context.log.warning("Couldn't find phone number")
```

Comment:
are we currently getting these warnings or is this just in case they change things in future?

Reply:
No, currently everyone has a e-mail and phone number. It's just in case it changes in the future

Reply:
Lovely, thanks!